### PR TITLE
update-cicd-pipeline-jdk-17

### PR DESCRIPTION
**Pull request by lucasdze**
This PR is to Update CI\CD Pipeline to use Java 17

### CODE REVIEW

**Review by jack-h-wang — CHANGES_REQUESTED** [review]
The action hash for sonarcloud needs to be updated as well.

File: prime-router/Dockerfile.build
Comment (jack-h-wang) [inline prime-router/Dockerfile.build:1]: Please leave updates to `prime-router` to the Platform team. Updates of this nature require more engineer coordination.

Comment (snesm) [reply prime-router/Dockerfile.build:1]: I believe that several changes here may affect prime-router.  Should these change be merged with any existing branch?

Comment (jack-h-wang) [reply prime-router/Dockerfile.build:1]: I don't think there needs to be any change to `prime-router` needed to migrate the CI/CD pipeline to Java 17. The work to make sure the codebase can be built to target Java 17 is a separate unit of work entirely. We just need the CI/CD pipeline to support Java 17 so this work can proceed.

File: .github/workflows/sonarcloud_master.yml
Comment (jack-h-wang) [inline .github/workflows/sonarcloud_master.yml:1]: Please update [the action hash for `sonarcloud-github-action`](https://github.com/CDCgov/prime-reportstream/blob/master/.github/workflows/sonarcloud_master.yml#L58).

File: prime-router/Dockerfile.build [hunk 1]
```diff
@@ -41,7 +41,7 @@ RUN rm Azure.Functions.Cli.linux-x64.${AFCT_VERSION}.zip
 # STAGE 2: Build the actual container that does the build
 #
 FROM debian:bullseye AS builder
-ARG JAVA_VERSION=11
+ARG JAVA_VERSION=17
```
Comment (jack-h-wang) [inline prime-router/Dockerfile.build:44]: Please revert this change for now.

File: prime-router/Dockerfile.dev [hunk 2]
```diff
@@ -5,8 +5,8 @@
 #
 # The intent is for the prime-router to be mounted into this container,
 # allowing the content of this container to be stable
-#
-ARG JAVA_VERSION=11
+
+ARG JAVA_VERSION=17
```
Comment (jack-h-wang) [inline prime-router/Dockerfile.dev:9]: Please revert this change for now.

**Review by jack-h-wang — CHANGES_REQUESTED** [review]
As previously commented, please leave changes to `prime-router` for the Platform team; the backend was building successfully on the updated CICD without this change, and any adjustments need to be coordinated among all engineering teams. Additionally, the SonarCloud Java 11 warning should no longer be appearing in comments.

File: prime-router/Dockerfile.build [hunk 3]
```diff
@@ -5,7 +5,7 @@
 #
 # --build-arg GRADLE_VERSION=7.6.1        - The version of Gradle you want to build with
 # --build-arg AFCT_VERSION=4.0.5198     - The version of the Azure Functions Core Tools
-# --build-arg JAVA_VERSION=11           - The version of the JDK (and thus JRE) you want to build against/with
+# --build-arg JAVA_VERSION=17           - The version of the JDK (and thus JRE) you want to build against/with
```
[the remark this answers is not in the document]
Comment (lucasdze) [reply prime-router/Dockerfile.build]: reverted

File: prime-router/Dockerfile.build (hunk 1 above)
Comment (lucasdze) [reply prime-router/Dockerfile.build]: reverted

File: prime-router/Dockerfile.dev (hunk 2 above)
Comment (lucasdze) [reply prime-router/Dockerfile.dev]: reverted

File: prime-router/Dockerfile.build
Comment (lucasdze) [reply prime-router/Dockerfile.build:1]: @jack-h-wang Please review the latest changes

File: .github/workflows/sonarcloud_master.yml
Comment (lucasdze) [reply .github/workflows/sonarcloud_master.yml:1]: @jack-h-wang 
We cant find a commit on sonarcloud-github-action action related to java 17. So we dont really know what hash to update to here. Any suggestion?

Comment (jack-h-wang) [reply .github/workflows/sonarcloud_master.yml:1]: @lucasdze Per the [update notes](https://community.sonarsource.com/t/java-11-is-deprecated-as-a-runtime-env-to-scan-your-projects/96597) it just needs to be a release tagged 2.0.0 or later.

Comment (snesm) [reply .github/workflows/sonarcloud_master.yml:1]: Long term statistics for SonarCloud Kotlin require that we build the back end and generate coverage reports even when building and testing the front end (building the back end does not require this).  SonarCloud uses the generated Jar and coverage reports for part of its testing. This Jar is built with a specific version of the JDK. We use the same JDK to run the scan (from setup-java).

**Comment by jack-h-wang** [comment]
Closing in favor of #10992.

**Pull request closed without merging**